chore: add cyclone web user guide (#1129)
caicloud/cyclone: docs/user-guide-web.md

> # Cyclone Web User Guide  

Reviewer: Use `doctoc` to generate an outline.

  
`Cyclone Web` provides a user friendly interface for users to interact with Cyclone. This document guides users how to use Cyclone web to build and execute workflows.

## Introduction

Cyclone Web consists of five components:

- Overview: Overview of the Cyclone platform
- Projects: Workflows are managed in projects, it's the main component of Cyclone web
- Templates: Stages templates for Cyclone
- Integrations: Integrated external services like Github, Docker registry.

> - Resource Types: Manage resource types Cyclone supported

Reviewer: ```suggestion
- Resource Types: Manage resource types Cyclone supports
```

### Overview

![overview](./images/screenshots/overview.png)

> ### Resource Types

Reviewer: I'd prefer that the order of component details keep consistent with that in Introduction.

Author: ok

> Resource in Cyclone is an important concept, it describes external data to be used as input/output to Cyclone workflows.

Reviewer: There is no need to emphasize the importance of `Resource` concept, as other concepts are also important.

```suggestion
Resource describes external data to be used as input/output of Cyclone workflows.
```

Users can edit existed integrations, or create new integrations here. For the moment, only some fixed type integrations (`SCM`, `DockerRegistry`, `SonarQube`, `Cluster`)  are supported in Cyclone web, an extensible way for general type integrations will come soon, please stay tuned.

### Templates

![templates](./images/screenshots/templates.png)

Stage templates helps Cyclone to fit in different kinds of scenes. A stage template describes how a stage works, and expose arguments to use it. Cyclone provides built-in templates for CI/CD scene:

- Build
- Unit Test
- Code Scan
- Image Build
- CD

A stage template defines input, output of a stage and expose necessary arguments for users to use the template.

![template detail](./images/screenshots/template-detail.png)

### Projects

Workflows in Cyclone are managed by projects. Project is a logical group of workflows. Users can define some common properties in project to be used by workflows belonging to it.

![project detail](./images/screenshots/project-detail.png)

3 kinds of resources are managed in project:

> - **Stages**  Building block of workflow. It defines how a stage works.

Reviewer: * Missing `:`
* `Building block` -> `Execution unit`

```suggestion
- **Stages**: Execution unit of workflow. It defines how a stage works.
```
